### PR TITLE
feat: add roundtrip test for textplans

### PR DESCRIPTION
**Pull request by EpsilonPrime**
The test is disabled by default as there are a large number of missing features and issues that still need to be addressed.

### CODE REVIEW

File: src/substrait/textplan/tests/CMakeLists.txt [hunk 1]
```diff
@@ -18,3 +18,52 @@ add_test_case(
...
+    POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E echo "Copying unit test data.."
+    COMMAND ${CMAKE_COMMAND} -E make_directory
+            "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/tests/data"
+    COMMAND
+      ${CMAKE_COMMAND} -E copy
+      "${TEXTPLAN_SOURCE_DIR}/converter/data/q6_first_stage.json"
+      "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/tests/data/q6_first_stage.json"
+    COMMAND ${CMAKE_COMMAND} -E copy "${TEXTPLAN_SOURCE_DIR}/data/*.json"
+            "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/tests/data/")
```
Comment (westonpace) [inline src/substrait/textplan/tests/CMakeLists.txt:56]: Is this needed for this test?

File: src/substrait/textplan/tests/CMakeLists.txt [hunk 2]
```diff
@@ -18,3 +18,52 @@ add_test_case(
...
+      "test data will be here:  ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/tests/data")
+else()
+
+  message(
+    STATUS
+      "Round trip testing is turned off.  Add SUBSTRAIT_CPP_ROUNDTRIP_TESTING=on to enable."
+  )
+
+endif()
+unset(SUBSTRAIT_CPP_ROUNDTRIP_TESTING CACHE)
```
Comment (westonpace) [inline src/substrait/textplan/tests/CMakeLists.txt:69]: Why is this needed?

File: src/substrait/textplan/tests/RoundtripTest.cpp [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            ParsesOk(),
+            HasErrors({}),
+            AsBinaryPlan(IgnoringFieldPaths(
+                {"extension_uris", "extensions"}, EqualsProto(plan)))))
+        << std::endl
+        << "Intermediate result:" << std::endl
+        << addLineNumbers(outputText);
+  } catch (std::invalid_argument ex) {
+    FAIL() << ex.what();
+  }
```
Comment (westonpace) [inline src/substrait/textplan/tests/RoundtripTest.cpp:90]: What is this catching?  Maybe use ASSERT_NO_THROW?

**Review by westonpace — COMMENTED** [review]
A few minor suggestions, but most of this seems to be test cases which is fine.

File: src/substrait/textplan/tests/CMakeLists.txt (hunk 1 above)
Comment (EpsilonPrime) [reply src/substrait/textplan/tests/CMakeLists.txt:56]: This copies all of the test data including the test plan used by the converter to be used by the roundtrip test.

File: src/substrait/textplan/tests/CMakeLists.txt (hunk 2 above)
Comment (EpsilonPrime) [reply src/substrait/textplan/tests/CMakeLists.txt:69]: No longer needed, removed.

File: src/substrait/textplan/tests/RoundtripTest.cpp (hunk 3 above)
Comment (EpsilonPrime) [reply src/substrait/textplan/tests/RoundtripTest.cpp:90]: Done.
